### ISSUE TITLE
[Bug]: Date axis labels formatted to show only the year are being localized as if they were numbers

### ISSUE THREAD

**Issue by krkshitij** [microsoft/fluentui#30128]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
    Memory: 18.41 GB / 31.71 GB
Browsers:
    Edge: Chromium (120.0.2210.77)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codepen.io/krkshitij/pen/yLZdGKp

## Actual Behavior
<img src="https://github.com/microsoft/fluentui/assets/110246001/ab30f2fc-6198-47b7-ae52-f1f766046fb8" alt="image" width="500" />

## Expected Behavior
The date x-axis in the above example should display the year as **1980** instead of **1,980**

### Requested priority

Normal

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by Shubhabrata08** [comment]
@krkshitij This is primarily because a Date Axis isn't configured for HeatMapChart and a String axis is fed by default to the xAxisType prop of CartesianChart
<img width="770" alt="image" src="https://github.com/microsoft/fluentui/assets/35366351/e97cd59a-db77-4e3f-a667-c312616de988">
<img width="519" alt="image" src="https://github.com/microsoft/fluentui/assets/35366351/63cac1b3-8ddd-4b6d-a718-34a3311d4b7b">

**Comment by krkshitij** [comment]
@Shubhabrata08 Yes, you are right. You can find some discussions about this issue [here](https://discord.com/channels/1158820827966480525/1158820827966480528/1189445284145668196).

But the fix seems to be a bit more involved. Currently, a string axis is created regardless of the data type, which uses the d3 [scaleBand](https://d3js.org/d3-scale/band) utility. This utility has a [bandwidth](https://d3js.org/d3-scale/band#band_bandwidth) property that is used to determine the width and height of rectangles. The d3 [scaleLinear](https://d3js.org/d3-scale/linear) and [scaleTime](https://d3js.org/d3-scale/time) utilities, used for number and date axes respectively, don't have such a property. We may need to figure out another way to resolve this issue.

**Comment by Shubhabrata08** [comment]
@krkshitij Is there a POC or draft under review currently regarding this issue?

**Comment by krkshitij** [comment]
@Shubhabrata08 No, I haven't seen any.

**Comment by harsh9975** [comment]
@AtishayMsft I found an approach to solve this issue.
 The label issue occurs only when we format data ``` xAxisDateFormatString="%Y"```
 So its formatted as string in `HeatMap` component and then it is converted as Number in Cartesian (since cartesian component is used by other components too its not a good approach to update the cartesian component)
 
 What we can do is whenever ``` xAxisDateFormatString="%Y"``` we can add a prefix "Y " this will stop cartesian component to localise the year. below is the example
 
<img width="732" alt="Screenshot 2024-04-26 at 2 24 58 AM" src="https://github.com/microsoft/fluentui/assets/33349461/7d7150cc-1cb0-4792-9c49-05030a6dc889">

Adding prefix can remove localisation of date as number and there is no need to update cartesian component
<img width="502" alt="Screenshot 2024-04-26 at 2 02 56 AM" src="https://github.com/microsoft/fluentui/assets/33349461/2cb61a2d-0fec-414d-b1c1-71062ad6c5c6">

 And further if we don't want the prefix to be printed we can just remove the "Y " from cartesian component and we will get the result
 
<img width="844" alt="Screenshot 2024-04-26 at 2 29 11 AM" src="https://github.com/microsoft/fluentui/assets/33349461/b0a6d97b-597c-41cc-ae63-6a28c33a39ca">

<img width="437" alt="Screenshot 2024-04-26 at 2 10 51 AM" src="https://github.com/microsoft/fluentui/assets/33349461/fd7031f5-ed0f-4b5c-88b2-6d5d3c7e2ac9">

This might not be the right approach.

**Comment by krkshitij** [comment]
keep open

**Comment by AtishayMsft** [comment]
This issue is fixed in vesion `5.23.85` of `react-charting` package.
